packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,558 @@
+
+  const { setPanelHeaderElements } = useSafeContext(PanelsWrapperContext);
+  const { id: panelId, associatedTrigger: panelAssociatedTrigger } =
+    useSafeContext(PanelContext);
+
+  const [element, setElement] = React.useState<HTMLElement | null>(null);
+
+  React.useEffect(() => {
+    setPanelHeaderElements((oldPanelHeaderElements) => {
+      if (oldPanelHeaderElements[panelId] === element) {
+        return oldPanelHeaderElements;
+      }
+
+      return {
+        ...oldPanelHeaderElements,
+        [panelId]: element,
+      };
+    });
+  }, [element, panelId, setPanelHeaderElements]);
+
+  const refs = useMergedRefs(setElement, forwardedRef);
+
+  return (
+    <Flex ref={refs} tabIndex={-1} {...rest}>
+      {panelAssociatedTrigger && <PanelBackButton />}
+      <Text id={`${panelId}-header-title`} as='h2'>
+        {children}
+      </Text>
+    </Flex>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+if (process.env.NODE_ENV === 'development') {
+  PanelHeader.displayName = 'Panels.Header';
+}
+
+// #endregion PanelHeader
+// ----------------------------------------------------------------------------
+// #region PanelBackButton
+
+/**
+ * Optional component that goes to the previous panel when clicked (i.e. to the panel that has a trigger that points to
+ * the current panel)
+ *
+ * @example
+ * <caption>Default back button</caption>
+ * <Panels.BackButton />
+ *
+ * @example
+ * <caption>Custom back button</caption>
+ * <Panels.BackButton onClick={() => console.log('Back button clicked')}>
+ *   <CustomBackIcon />
+ * </Panels.BackButton>
+ */
+const PanelBackButton = React.forwardRef((props, forwardedRef) => {
+  const { children, onClick, ...rest } = props;
+
+  const { instance: panelInstance } = useSafeContext(PanelsInstanceContext);
+
+  return (
+    <IconButton
+      ref={forwardedRef}
+      label='Back'
+      styleType='borderless'
+      size='small'
+      data-iui-shift='left'
+      {...rest}
+      onClick={mergeEventHandlers(
+        React.useCallback(() => panelInstance?.goBack(), [panelInstance]),
+        onClick,
+      )}
+    >
+      {children || <SvgChevronLeft />}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<'button', IconButtonProps>;
+if (process.env.NODE_ENV === 'development') {
+  PanelBackButton.displayName = 'Panels.BackButton';
+}
+
+// #endregion PanelBackButton
+// ----------------------------------------------------------------------------
+
+export const Panels = {
+  Wrapper: PanelsWrapper,
+  Panel,
+  Trigger: PanelTrigger,
+  Header: PanelHeader,
+  useInstance: useInstance as () => PanelsInstance,
+};
+
+/**
+ * Returns a copy of value which reflects state changes after a set delay.
+ */
+function useDelayed<T>(
+  value: T,
+  {
+    delay,
+  }: {
+    /** Default delay is 500ms, or 0ms when prefers-reduced-motion */
+    delay?: number;
+  } = {},
+): T {
+  const motionOk = getWindow()?.matchMedia?.(
+    '(prefers-reduced-motion: no-preference)',
```

Comment:
> So where do you suggest I create `motionOk`? In the components that call `useDelayed()` and then make `delay` a required parameter, or something along those lines?

It should be called only where it's needed. In this case, the main `Panels` component would call it, since it only needs to be used for changing the `scrollIntoView` `behavior`. I think changing the `delay` might not be necessary (unconditionally leaving the panel in the DOM for 500ms should not cause any adverse effect).

> So, should we replace the `getWindow()?.matchMedia?.('(prefers-reduced-motion: no-preference)',)?.matches` calls in other components with `useMediaQuery()` in a future PR?

Yeah that makes sense to me.

Reply:
> Sounds good, made that change ([c296e05](https://github.com/iTwin/iTwinUI/commit/c296e0541cbb416cad8e52bcd0d5120c01331e32)).

Not sure if it's on purpose but the `delay` is now `5000` (instead of `500`) and the initial value returned by `useDelayed` is no longer `undefined`.